TelCo.ColorCoder/Base/ColorProcessor.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using TelCo.ColorCoder.Interface;
+using TelCo.ColorCoder.UnitTestAssertion;
+
+namespace TelCo.ColorCoder
+{
+    public class ColorProcessor
+    {
+        private ColorPair _pair;
+        private int _pairNumber, minorSize;
+        public ColorProcessor(ColorPair pair = null, int pairNumber = 0)
+        {
+            _pair = pair;
+            _pairNumber = pairNumber;
+        }
+        public void GetPairNumberFromColorProcessor()
+        {
+            int majorIndex = ColorCalculationForPairNumberFromColor.GetMajorColor(_pair);
+            int minorIndex = ColorCalculationForPairNumberFromColor.GetMinorColor(_pair);
+            int result = EvaluateColor.EvaluateColorPairNumber(_pair, majorIndex, minorIndex);
+            Console.WriteLine("[In]Colors: {0}, [Out] PairNumber: {1}\n", _pair, result);
+            IUnitTestForColor unitTestForColor = new PairNumberFromColorAssertion(_pair, result);
+            unitTestForColor.AssertionCheck();
```

Comment:
Assertion check is part of production code here. Test code should be separate. It should use production code and verify its output. Here production code uses test code. 
This assertion check can cause issues in field !

To test production code independently and effectively, it is always better to keep these codein separate files.

Reply:
Yes, correct I should add the proper Unit test project under solution and write unit test methods for respective action methods going forward I'll try to implement Xunit/Nunit unit test project into project and achieve the same.

Thanks for your valuable comments.

Reply:
I have added separate unit test project for testing 👍